### PR TITLE
test structure is introduced in pytest

### PR DESCRIPTION
**Pull request by AhmetCanSolak**
# What does this PR do? Why are doing this? References?
Test structure was revised with @kne42  and @royerloic , here is updated introduction of test structure in `pytest`.

## Type of change
[Please delete options that are not relevant.]
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
This commit consist of test files.

## Final Checklist:
- [X] My PR is the possible minimum work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by royerloic — APPROVED** [review]
Great!

File: gui/util/test_misc.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import pytest
+
+import misc
```
Comment (kne42) [inline gui/util/test_misc.py:3]: relative imports please

**Review by kne42 — COMMENTED** [review]
Would prefer structure to look like `{module}/tests/test_{file}.py` instead of `{module}/test_{file}.py` to keep tests separate and "hidden" to an extent.

**Comment by jni** [comment]
I agree with @kne42, tests should be in a separate directory. See the scikit-image codebase for an example.

Can you explain what is going on with the meta dict? Where will this be used? I want to make sure that we don't deviate too far from (and certainly stay compatible with) imageio.

**Comment by kne42** [comment]
@jni That was really there just for demo purposes since we don't actually have a standard for metadata yet.

**Pull request merged**